src/Common/src/CoreLib/Interop/Unix/System.Native/Interop.GetTimestamp.cs

```diff
@@ -8,10 +8,10 @@ internal static partial class Interop
 {
     internal static partial class Sys
     {
-        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetTimestampResolution")]
-        internal static extern bool GetTimestampResolution(out long resolution);
+        [DllImport(Libraries.SystemNative, BestFitMapping = false, CallingConvention = CallingConvention.Cdecl, EntryPoint = "SystemNative_GetTimestampResolution", ExactSpelling = true, PreserveSig = true, SetLastError = false, ThrowOnUnmappableChar = false)]
+        internal static extern unsafe int GetTimestampResolution(long* resolution);
 
-        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetTimestamp")]
-        internal static extern bool GetTimestamp(out long timestamp);
+        [DllImport(Libraries.SystemNative, BestFitMapping = false, CallingConvention = CallingConvention.Cdecl, EntryPoint = "SystemNative_GetTimestamp", ExactSpelling = true, PreserveSig = true, SetLastError = false, ThrowOnUnmappableChar = false)]
```

Comment:
Aren't most of these values defaults? We've not been specifying defaults in general. If we're going to start, it should be done everywhere, not just one place.

Reply:
`CallingConvention`, `EntryPoint`, `PreserveSig`, `SetLastError`, and `ThrowOnUnmappableChar` and have their existing defaults.

`CallingConvention` varies by platform, so I think it is good to set here explicitly.
`SetLastError`, in this case, is being explicitly set to false even though the underlying method does set it, since we never check the last error.

I don't feel particularly strongly about the other three, but I have always found being explicit about P/Invoke signatures to be the best solution as it avoids cross-platform or cross-runtime differences and makes the intent of the invocation very clear.

Reply:
> I have always found being explicit about P/Invoke signatures to be the best solution as it avoids cross-platform or cross-runtime differences and makes the intent of the invocation very clear.

While I hear you on that, I've found the opposite: setting all of the values makes it long list of gobbledygook where it's hard to see the signal in the noise.  When you only set the things that differ from the defaults, you can learn the defaults and then clearly see exactly what's different on any given signature.  When you instantiate a class or struct, you rarely go through and set every settable property, or rarely use the longest possible constructor passing in default values for everything; instead you rely on the defaults and only set the things you need to set. 

> CallingConvention varies by platform

And the default maps to the platform.

Reply:
I agree with Stephen. It is makes the code very hard to read if you specify all default values. 

The general CoreFX style for PInvokes is to specify only what is required.

Reply:
Fixed.